lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,128 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = [];
+			$function = function (IUser $user) use (&$users) {
+				$users[] = $user;
+			};
+
+			$this->userManager->callForAllUsers($function);
```

Comment:
This is bad. You shouldn't use the callback to collect all the users. The purpose of using a callback is to avoid this precisely, to avoid having a list of all users in memory.
```
$this->userManager->callForAllUsers(function(IUser $user) use ($input, $output) {
  $this->setPasswordExpireForUser($input, $output, $user);
});
```
Note that the callback function process only one user.

Reply:
Updated `callForUsers` as per the comment.